app/src/test/java/org/vss/api/GetObjectApiTest.java

```diff
@@ -0,0 +1,87 @@
+package org.vss.api;
+
+import com.google.protobuf.ByteString;
+import jakarta.ws.rs.core.Response;
+import java.nio.charset.StandardCharsets;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.vss.ErrorCode;
+import org.vss.ErrorResponse;
+import org.vss.GetObjectRequest;
+import org.vss.GetObjectResponse;
+import org.vss.KVStore;
+import org.vss.KeyValue;
+import org.vss.exception.ConflictException;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class GetObjectApiTest {
+  private GetObjectApi getObjectApi;
+  private KVStore mockKVStore;
+
+  private static String TEST_STORE_ID = "storeId";
+  private static String TEST_KEY = "key";
+  private static KeyValue TEST_KV = KeyValue.newBuilder().setKey(TEST_KEY).setValue(
+      ByteString.copyFrom("test_value", StandardCharsets.UTF_8)).build();
+
+  @BeforeEach
+  void setUp() {
+    mockKVStore = mock(KVStore.class);
+    getObjectApi = new GetObjectApi(mockKVStore);
+  }
+
+  @Test
+  void execute_ValidPayload_ReturnsResponse() {
```

Comment:
i tried creating a common base testcase but it was more complex and less readable since many api specific calls are interleaved.

Reply:
these are the abstract methods a sub-test would have had to implement in that case, and it seemed difficult to understand test-case.
```
  abstract void beforeEachSetup();
  
  abstract void verifyAfterTest();

  abstract GeneratedMessageV3 createExpectedRequest();

  abstract GeneratedMessageV3 createExpectedResponse();
  
  abstract Response executeValidRequest();

  abstract Response executeErroringRequest(Class<? extends Exception> exceptionClass)
      throws Exception;
      
```